FreeRTOS-Plus/Demo/FreeRTOS_Cellular_Interface_Windows_Simulator/Common/cellular_setup.c

```diff
@@ -131,27 +127,55 @@ bool setupCellular( void )
 
             vTaskDelay( pdMS_TO_TICKS( CELLULAR_SIM_CARD_WAIT_INTERVAL_MS ) );
         }
+
+        if( cellularStatus != CELLULAR_SUCCESS )
+        {
+            configPRINTF( ( ">>>  Cellular SIM failure  <<<\r\n" ) );
+        }
+    }
+
+    /* Turn off PSM because this is demo to showcase MQTT instead of PSM mode. */
+    if( cellularStatus != CELLULAR_SUCCESS )
+    {
+        psmSettings.mode = 0;
+        cellularStatus = cellularStatus = Cellular_SetPsmSettings( CellularHandle, &psmSettings );
+
+        if( cellularStatus != CELLULAR_SUCCESS )
+        {
+            configPRINTF( ( ">>>  Cellular_SetPsmSettings failure  <<<\r\n" ) );
+        }
     }
 
     /* Setup the PDN config. */
     if( cellularStatus == CELLULAR_SUCCESS )
     {
         cellularStatus = Cellular_SetPdnConfig( CellularHandle, CellularSocketPdnContextId, &pdnConfig );
-    }
-    else
-    {
-        configPRINTF( ( ">>>  Cellular SIM failure  <<<\r\n" ) );
+
+        if( cellularStatus != CELLULAR_SUCCESS )
+        {
+            configPRINTF( ( ">>>  Cellular_SetPdnConfig failure %d  <<<\r\n", cellularStatus ) );
+        }
     }
 
     /* Rescan network. */
     if( cellularStatus == CELLULAR_SUCCESS )
     {
         cellularStatus = Cellular_RfOff( CellularHandle );
+
+        if( cellularStatus != CELLULAR_SUCCESS )
+        {
+            configPRINTF( ( ">>>  Cellular_RfOff failure %d  <<<\r\n", cellularStatus ) );
+        }
     }
 
     if( cellularStatus == CELLULAR_SUCCESS )
     {
         cellularStatus = Cellular_RfOn( CellularHandle );
+
+        if( cellularStatus != CELLULAR_SUCCESS )
+        {
+            configPRINTF( ( ">>>  Cellular_RfOn failure %d  <<<\r\n", cellularStatus ) );
+        }
     }
 
```

Comment:
Is registration always automatic when RF is enabled?

Reply:
Cellular interface library reply on automatic registration by modem. We can consider to add this feature to select PLMN manually.